addons/pinniped/post-deploy/pkg/configure/configure_test.go

```diff
@@ -14,18 +14,314 @@ import (
+				// therefore:
+				// - audience=amazing.cluster-1234567890 is set as Audience
+				// - Cluster.Type is set as "workload" which is passed down into Pinniped()
+				// unfortunately our assertions here do not make this behavior completely clear
+				vars.JWTAuthenticatorAudience = "amazing.cluster-1234567890"
+			},
+			cleanup: func() {
+				vars.JWTAuthenticatorAudience = ""
+			},
+		},
+		// TODO: this test exercises a code path that is partially implemented and not exercised today.
+		// When TKGm removes the tkg-metadata configmap this path will be executed.
+		// Today it will error (correctly) as we can't recover in this state.
+		// {
+		// 	name: "When the tkg-metadata configmap is not found, nodes are queried for the xcluster.x-k8s.io/cluster-name annotation but no cluster.type is used (PARTIAL IMPL, THIS IS BROKEN LONG TERM)",
+		// 	newKubeClient: func() *kubefake.Clientset {
+		// 		c := kubefake.NewSimpleClientset(
+		// 			// a variety of objects are needed to fake out various helpers within the TKGAuthentication() chain
+		// 			// tkgMetadataConfigmapManagementCluster, <- nope, not here this time
+		// 			conciergeDeployment,
+		// 			supervisorDeployment,
+		// 			supervisorService,
+		// 			supervisorCertificateSecret,
+		// 		)
+		// 		c.PrependReactor("delete", "secrets", func(action kubetesting.Action) (bool, runtime.Object, error) {
+		// 			// When we delete the secret in the implementation, we expect the cert-manager controller
+		// 			// to recreate it. Since there is no cert-manager controller running, let's just tell the
+		// 			// fake kube client to not delete it (i.e., that we "handled" the delete ourselves).
+		// 			return actionIsOnObject(action, supervisorCertificateSecret), nil, nil
+		// 		})
+		// 		return c
+		// 	},
+		// 	newCertManagerClient: func() *certmanagerfake.Clientset {
+		// 		return certmanagerfake.NewSimpleClientset(supervisorCertificate)
+		// 	},
+		// 	newSupervisorClient: func() *pinnipedsupervisorfake.Clientset {
+		// 		// return a fake idp when requsted
+		// 		return pinnipedsupervisorfake.NewSimpleClientset(upstreamIdentityProvider)
+		// 	},
+		// 	newConciergeClient: func() *pinnipedconciergefake.Clientset {
+		// 		defaultJWTAuthenticator := jwtAuthenticator.DeepCopy()
+		// 		return pinnipedconciergefake.NewSimpleClientset(defaultJWTAuthenticator)
+		// 	},
+		// },
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			fakeKubeClient := test.newKubeClient()
+			fakeCertManagerClient := test.newCertManagerClient()
+			supervisorClientset := test.newSupervisorClient()
+			conciergeClientset := test.newConciergeClient()
+			clients := Clients{
+				K8SClientset:         fakeKubeClient,
+				CertmanagerClientset: fakeCertManagerClient,
+				SupervisorClientset:  supervisorClientset,
+				ConciergeClientset:   conciergeClientset,
+			}
+			if test.finalSetup != nil {
+				test.finalSetup()
+			}
+			err := TKGAuthentication(clients)
```

Comment:
This is a start, but we really should test for more than just if `TKGAuthentication()` errors or not.
Did the JWTAuthenticator get configured correctly?

Reply:
Positive assertions would be better than just negative assertions.

Reply:
The meat of this is that the JWTAuthenticator is correctly configured with the correct audience.
The `Cluster.Type` value is needed to deploy Pinniped Supervisor (or not).
So a stronger test assertion would check for these things to be correct.

Reply:
revised.